wpilibj/src/test/java/edu/wpi/first/wpilibj/hal/MatchInfoDataTest.java

```diff
@@ -9,13 +9,31 @@
 
 import org.junit.jupiter.api.Test;
 
+import edu.wpi.first.hal.sim.mockdata.DriverStationDataJNI;
+import edu.wpi.first.wpilibj.DriverStation.MatchType;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
 class MatchInfoDataTest {
   @Test
-  @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
   void matchInfoDataDoesNotThrow() {
     HAL.initialize(500, 0);
-    MatchInfoData data = new MatchInfoData();
-    HAL.getMatchInfo(data);
-    // Nothing we can assert, so just make sure it didn't throw.
+
+    MatchInfoData inMatchInfo = new MatchInfoData();
+    inMatchInfo.eventName = "Event Name";
+    inMatchInfo.gameSpecificMessage = "Game Message";
+    inMatchInfo.matchNumber = 174;
+    inMatchInfo.matchType = MatchType.Qualification.ordinal();
+    inMatchInfo.replayNumber = 191;
+    DriverStationDataJNI.setMatchInfo(inMatchInfo);
+
+    MatchInfoData outMatchInfo = new MatchInfoData();
+    HAL.getMatchInfo(outMatchInfo);
+
+    assertEquals("Event Name", outMatchInfo.eventName);
+    assertEquals(MatchType.Qualification.ordinal(), outMatchInfo.matchType);
+    assertEquals(174, outMatchInfo.matchNumber);
+    assertEquals(191, outMatchInfo.replayNumber);
+    assertEquals("Game Message", outMatchInfo.gameSpecificMessage);
```

Comment:
Each of these should be their own test.

Reply:
That seems unnecessarily verbose and isn't the case with multiple other tests in allwpilib

Reply:
Yes... I am actively working on fixing that.  What ends up happening is a few years will go by and no one will know why the test broke or what it does.  If we follow the one assert per test, what each test is doing is extremely clear.  Just as an example, look at this test: [CommandSequentialGroupTest.java](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/test/java/edu/wpi/first/wpilibj/command/CommandSequentialGroupTest.java). When that test fails don't even try to figure out what happened.

Reply:
You could even automate the test... one moment.

Reply:
Actually... never mind on the automating.  It is too complex with multiple types.